compiler/data/class-data.h

```diff
@@ -80,6 +84,8 @@ class ClassData : public Lockable {
   FFIClassDataMixin *ffi_class_mixin = nullptr; // non-null for ffi_cdata classes
   FFIScopeDataMixin *ffi_scope_mixin = nullptr; // non-null for ffi_scope classes
 
+  std::set<ClassPtr> json_encoders;
```

Comment:
Just a note to consider: sizeof empty std::set on 64-bit machine is 48 bytes.
I believe this set will be empty for ~95% of classes and will contain 2-4 elements at max in the average case.
Previously, we used forward lists (8 bytes) of key-values for these cases previously, so for every class we allocate less extra bytes unless we need them. This is important since for bigger projects we have thousands of classes and we may hit the memory threshold unless we're careful.